### PR TITLE
Dynamic dropdowns

### PR DESCRIPTION
**Pull request by shulijones**
(not yet deployed)
When a user picks one dropdown option, exclude others that are not relevant (e.g. if they pick 'children', exclude 'work').

### CODE REVIEW

File: src/main/webapp/query-control.js [hunk 1]
```diff
@@ -290,6 +290,52 @@ function replaceValueIfEmpty(dataListId) {
...
+    options = defaultDropdowns.get('person-type');
+  } else {
+    dropdown = document.getElementById('action');
+    options = defaultDropdowns.get('action');
+  }
+  dropdown.innerHTML = '';
+  const currentValue = document.getElementById(type + '-list').value;
+
+  options.forEach((option) => {
+    if (!(dropdownsToExclude.has(currentValue) &&
```
Comment (kyracq) [inline src/main/webapp/query-control.js:331]: I think a comment explaining this if statement would be helpful

Comment (shulijones) [reply src/main/webapp/query-control.js:331]: Sure, good point! Added.

File: src/main/webapp/query-control.js [hunk 2]
```diff
@@ -290,6 +290,56 @@ function replaceValueIfEmpty(dataListId) {
   }
 }
 
+// When the user picks certain dropdown values,
+// exclude the combinations that don't have data available
+const dropdownsToExclude = new Map();
+dropdownsToExclude.set(
```
Comment (nsri-google) [inline src/main/webapp/query-control.js:296]: the same values have to be listed twice once for persontype to action and once again for action to persontype. You could instead make this just a set of personType+action. Then when checking you compute personType+action and check if it exists in the set.

File: src/main/webapp/query-control.js [hunk 3]
```diff
@@ -290,6 +290,56 @@ function replaceValueIfEmpty(dataListId) {
...
+  }
+  dropdown.innerHTML = '';
+  const currentValue = document.getElementById(type + '-list').value;
+
+  // Iterate through all potential options
+  options.forEach((option) => {
+    // Check if the current option is in the exclusions list for the value that
+    // the user entered; if it's not, add it to the dropdown
+    if (!(dropdownsToExclude.has(currentValue) &&
+        dropdownsToExclude.get(currentValue).includes(option[0]))) {
```
[the remark this answers is not in the document]
Comment (shulijones) [reply src/main/webapp/query-control.js:336]: Good point. Done

File: src/main/webapp/query-control.js (hunk 2 above)
Comment (shulijones) [reply src/main/webapp/query-control.js:296]: Ah, good idea. I've done this, hopefully in a way that's not too clunky.

**Pull request merged**